scalding-spark/src/test/scala/com/twitter/scalding/spark_backend/SparkBackendTests.scala

```diff
@@ -201,3 +203,34 @@ class SparkBackendTests extends FunSuite with BeforeAndAfter {
       Config.empty.setForceToDiskPersistMode("NONE"))
   }
 }
+
+class ConfigPartitionComputerTest extends PropSpec with PropertyChecks {
+  property("when no config or number of reducers are given, returns the current number of partitions (or 1)") {
+    val pc = ConfigPartitionComputer(Config.empty, None)
+    forAll { i: Int =>
+      if (i >= 1) assert(pc(i) == i)
+      else assert(pc(i) == 1)
+    }
+  }
+
+  property("when number of reducers are given but no scaling factor, returns the number of reducers") {
+    val pc = ConfigPartitionComputer(Config.empty, Some(10))
+    forAll { i: Int =>
+      pc(i) == 10
+    }
+  }
+
+  property("when reducer scaling factor given, scales the number of reducers") {
+    val pc = ConfigPartitionComputer(Config.empty.setReducerScaling(2.0), Some(10))
+    forAll { i: Int =>
+      pc(i) == 2.0D * 10
```

Comment:
I think this should be `20` not since `2.0D * 10` is a double, right, but pc is always returning an Int?